lib/rspec/mocks/method_double.rb

```diff
@@ -44,6 +43,8 @@ def visibility
 
       # @private
       def object_singleton_class
+        # We can't use @object.singleton_class because this method
+        # creates a new singleton class if the object doesn't have one.
```

Comment:
Doesn't this apply to the previous swaps?

Reply:
This statement still applies to other cases where I've replaced `(class << ... ; self; end)` but leaves specs green.
Do you think it makes sense to use our implementation of `singleton_method` that doesn't accidentally create a singleton class if it doesn't exist?
Would you consider adding a helper to `rspec-support`? AFAIR, we use this quite extensively here and in `rspec-expectations`.

Reply:
I'd just revert the other changes, it's a well known technique